test/scenarios.js

```diff
@@ -0,0 +1,1252 @@
+    } = await setupContracts());
+    signers = await ethers.getSigners();
+    blockReward = await blockManager.blockReward();
+  });
+
+  beforeEach(async () => {
+    snapShotId = await takeSnapshot();
+    await Promise.all(await initializeContracts());
+    await assetManager.grantRole(ASSET_MODIFIER_ROLE, signers[0].address);
+    await governance.grantRole(GOVERNER_ROLE, signers[0].address);
+    const url = 'http://testurl.com';
+    const selector = 'selector';
+    const selectorType = 0;
+    let name;
+    const power = -2;
+    const weight = 50;
+    let i = 0;
+    while (i < 9) {
+      name = `test${i}`;
+      await assetManager.createJob(weight, power, selectorType, name, selector, url);
+      i++;
+    }
+
+    while (Number(await stakeManager.getState(await stakeManager.epochLength())) !== 4) { await mineToNextState(); }
+
+    let Cname;
+    for (let i = 1; i <= 8; i++) {
+      Cname = `Test Collection${String(i)}`;
+      await assetManager.createCollection([i, i + 1], 1, 3, Cname);
+    }
+    Cname = 'Test Collection9';
+    await assetManager.createCollection([9, 1], 1, 3, Cname);
+
+    await mineToNextEpoch();
+    const epoch = getEpoch();
+    const razors = tokenAmount('443000');
+
+    await razor.transfer(signers[1].address, razors);
+    await razor.transfer(signers[2].address, razors);
+    await razor.transfer(signers[3].address, razors);
+    await razor.transfer(signers[4].address, razors);
+    await razor.transfer(signers[5].address, razors);
+
+    let stake = razors.sub(tokenAmount(Math.floor((Math.random() * 442000))));
+
+    await razor.connect(signers[1]).approve(stakeManager.address, stake);
+    await stakeManager.connect(signers[1]).stake(epoch, stake);
+    stakes.push(stake);
+
+    stake = razors.sub(tokenAmount(Math.floor((Math.random() * 442000))));
+    await razor.connect(signers[2]).approve(stakeManager.address, stake);
+    await stakeManager.connect(signers[2]).stake(epoch, stake);
+    stakes.push(stake);
+
+    stake = razors.sub(tokenAmount(Math.floor((Math.random() * 442000))));
+    await razor.connect(signers[3]).approve(stakeManager.address, stake);
+    await stakeManager.connect(signers[3]).stake(epoch, stake);
+    stakes.push(stake);
+
+    stake = razors.sub(tokenAmount(Math.floor((Math.random() * 442000))));
+    await razor.connect(signers[4]).approve(stakeManager.address, stake);
+    await stakeManager.connect(signers[4]).stake(epoch, stake);
+    stakes.push(stake);
+
+    stake = razors.sub(tokenAmount(Math.floor((Math.random() * 442000))));
+    await razor.connect(signers[5]).approve(stakeManager.address, stake);
+    await stakeManager.connect(signers[5]).stake(epoch, stake);
+    stakes.push(stake);
+  });
+
+  afterEach(async () => {
+    await restoreSnapshot(snapShotId);
+    stakes = [];
+  });
+
+  it('100 epochs of constant voting and participation', async () => {
```

Comment:
Are we going to run scenario tests per commit in CI, or have periodic runs ?
If we do every time CI would take significant time, 2X with gasCompare

Reply:
As of right now, it is running every commit we make. I do agree though, CI would take a lot of time with currently how it is